components/auth/SignUpForm.tsx

```diff
@@ -24,37 +24,26 @@ const SignUpForm = () => {
     resolver: zodResolver(signUpSchema),
     defaultValues: {
       fullname: "",
-      username: "",
+      email: "",
       password: "",
     },
   });
 
   const onSubmit = async () => {
-    const { fullname, username, password } = form.getValues();
+    const { email, password } = form.getValues();
 
     try {
       startTransition(async () => {
-        const res = await fetch("/api/register", {
-          method: "POST",
-          headers: {
-            "Content-Type": "application/json",
-          },
-          body: JSON.stringify({ fullname, username, password }),
+        await signIn("credentials", {
+          redirect: true,
+          email,
+          password,
+          callbackUrl: "/",
```

Comment:
This is your signup form. You're just trying to sign in. You need to call your server-action for create user here. You have a createNewUser function in `user.actions.ts`.

Slow down. Get the issue fixed, and then test it out to make sure it works. One step at a time 👍

Reply:
👍

Reply:
`const onSubmit = async () => {
    const { fullname, email, password } = form.getValues();

    try {
      startTransition(async () => {
        const res = await createNewUser({ fullname, email, password });

        if (res?.ok)
          await signIn("credentials", {
            redirect: true,
            email,
            password,
            callbackUrl: `${window.location.origin}`,
          });
      });
    } catch (err: any) {
      setMessage(err);
      setFormError(err);
    }
  };` - onSubmit function on signUpPage